src/main/java/com/gluonhq/substrate/util/windows/MSIBundler.java

```diff
@@ -0,0 +1,325 @@
+        if (Files.exists(userAssets)) {
+            FileOps.copyDirectory(userAssets, windowsAssetPath);
+        }
+
+        Path config = tmpMSI.resolve("config");
+        Files.createDirectories(config);
+        Path wixPath = config.resolve("main.wxs");
+        Path wixObjPath = wixPath.getParent().resolve(wixPath.getFileName() + ".wixobj");
+        Path msiPath = paths.getGvmPath().getParent().resolve(appName + "-" +  version  + ".msi");
+        FileOps.copyResource("/native/windows/wix/main.wxs", wixPath);
+        FileOps.copyDirectory(windowsAssetPath, config);
+
+        /**
+         * Wix Compile
+         */
+        Map<String, String> userInput = createAppDetailMap();
+        List<String> processArgs = new ArrayList<>(List.of(
+                WixTool.CANDLE.getPath(),
+                "-nologo",
+                wixPath.toString(),
+                "-ext", "WixUtilExtension",
+                "-arch", "x64",
+                "-out", wixObjPath.toString()
+        ));
+
+        userInput.entrySet().stream()
+                .map(wixVar -> String.format("-d%s=%s", wixVar.getKey(), wixVar.getValue()))
+                .forEachOrdered(processArgs::add);
+        Logger.logInfo(String.join(" ", processArgs));
+        ProcessRunner candle = new ProcessRunner(processArgs.toArray(new String[0]));
+        if (candle.runProcess("Wix Compiler") != 0) {
+            throw new IOException("Error running candle to generate wixobj");
+        }
+
+        /**
+         * Wix Link
+         */
+        processArgs.clear();
+        processArgs.addAll(List.of(
+                WixTool.LIGHT.getPath(),
+                "-nologo",
+                "-spdb",
+                "-sw1076", // https://github.com/wixtoolset/issues/issues/5938
+                "-ext", "WixUtilExtension",
+                "-ext", "WixUIExtension",
+                "-out", msiPath.toString(),
+                wixObjPath.toString()
+        ));
+
+        ProcessRunner light = new ProcessRunner(processArgs.toArray(new String[0]));
+        if (light.runProcess("Wix Linker") != 0) {
+            throw new IOException("Error running light to generate msi");
+        }
+
+        return true;
+    }
+
+    private Map<String, String> createAppDetailMap() {
+
+        Map<String, String> userInput = new HashMap<>();
+        ReleaseConfiguration releaseConfiguration = projectConfiguration.getReleaseConfiguration();
+
+        String appName = projectConfiguration.getAppName();
+        String executableName = appName + ".exe";
+        String vendor = Optional.ofNullable(releaseConfiguration.getVendor()).orElse("Unknown");
+        String version = Optional.ofNullable(releaseConfiguration.getVersionName()).orElse(DEFAULT_APP_VERSION);
+        userInput.put("GSProductCode", createUUID("ProductCode", appName, vendor, version).toString());
+        userInput.put("GSAppName", appName);
```

Comment:
If `appName` contains spaces, `<File Id="$(var.GSAppName)"` fails with:
```
error CNDL0014 : The File/@Id attribute's value, '"My Gluon Application"', is not a legal identifier.  Identifiers may contain ASCII characters A-Z, a-z, digits, underscores (_), or periods (.).  Every identifier must begin with either a letter or an underscore.
```
Maybe you need some `GSAppNameId` where you use `appName.replaceAll("\\s","_")`

Reply:
Isn't it the responsibility of the user to provide the correct app name?

Reply:
hmm, we use for the app name the `<name/>` from the Maven [project](https://github.com/gluonhq/gluonfx-maven-plugin/blob/master/src/main/java/com/gluonhq/NativeBaseMojo.java#L202), which is "the full name of the project", so I take it can have spaces.

Dealing with this is cumbersome but, unless we force a check on the appName to not contain spaces, we must take it into account in Substrate.

Reply:
Fixed this with recent commit